pkg/resource/deploy/step.go

```diff
@@ -25,9 +25,19 @@ import (
 	"github.com/pulumi/pulumi/pkg/util/contract"
 )
 
+// StepCompleteFunc is the type of functions returned from Step.Apply. These functions are to be called
+// when the engine has fully retired a step.
+type StepCompleteFunc func()
```

Comment:
If we're going to use the terms `retired` or `committed` to refer to persisting the results of a step to the snapshot, how about `StepCommittedFunc` or `StepRetiredFunc`?

Reply:
FWIW I think that `complete` is better jargon because a step can be `preview`'d. In that case "completion" makes more sense to me because there is nothing to commit.